src/components/Coins.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { getCoins } from '../redux/coins/CoinsSlice';
+
+const Coins = () => {
+  const { coins, isLoading, errorMessage } = useSelector((state) => state.coins);
+  const dispatch = useDispatch();
+  const [searchText, setSearchText] = useState('');
+
+  useEffect(() => {
+    dispatch(getCoins());
+  }, [dispatch]);
+
+  const handleSearch = (e) => {
+    setSearchText(e.target.value);
+  };
+
+  const filteredCoins = coins.filter(
+    (coin) => coin.name.toLowerCase().includes(searchText.toLowerCase()),
+  );
+
+  return (
+    <section className="coin-section">
+      <div className="coin-topic">
+        <h2>Cryptocurrency Market</h2>
+        <p>
+          (
+          {filteredCoins.length}
+          {' '}
+          Famouse Crypto Coins)
+        </p>
+        <div className="search-bar-container">
+          <input type="text" placeholder="Search" className="search-bar" value={searchText} onChange={handleSearch} />
+        </div>
+      </div>
+      {isLoading && <h2 style={{ color: '#fff', margin: 'auto' }}>Loading...</h2>}
+      {errorMessage && <h2 style={{ color: '#fff', margin: 'auto' }}>{errorMessage}</h2>}
+
+      <div className="card-container">
+        {filteredCoins.map((coin) => (
+          <Link to={`/details/${coin.id}`} key={coin.id} className="coin-card">
+            <div>
+              <h3>{coin.name}</h3>
+              <p>
+                {parseFloat(coin.priceUsd).toFixed(2)}
+                {' '}
+                USD
+              </p>
+            </div>
+          </Link>
```

Comment:
- [ ] Please add some `space` around the `text` of your cards, and as per design, you should also add the `arrow` icon, as it is given in the actual design for `navigation` purposes. As we (you and the other micronauts) know if we click on the card then we will see the `detail` page of that coin. But, think about the external people who are not micronauts, take a `technical` recruiter as an example who wants to hire you and look into your project, then it would be convenient for him/her to navigate easily from one page to another through some hints, which in this case is `arrow` icon.

![image](https://github.com/tawakuliKH/coinCap-React/assets/66476329/37f6547b-9292-4d94-8a21-91aa5ce72596)